nest-tax-backend/src/tax/utils/unified-tax.util.ts

```diff
@@ -0,0 +1,406 @@
+import { TaxDetail } from '@prisma/client'
+import dayjs, { Dayjs } from 'dayjs'
+import timezone from 'dayjs/plugin/timezone'
+import utc from 'dayjs/plugin/utc'
+
+import ThrowerErrorGuard from '../../utils/guards/errors.guard'
+import {
+  QrCodeGeneratorDto,
+  QrPaymentNoteEnum,
+} from '../../utils/subservices/dtos/qrcode.dto'
+import {
+  CustomErrorTaxTypesEnum,
+  CustomErrorTaxTypesResponseEnum,
+} from '../dtos/error.dto'
+import {
+  InstallmentPaidStatusEnum,
+  InstallmentPaymentReasonNotPossibleEnum,
+  OneTimePaymentReasonNotPossibleEnum,
+  OneTimePaymentTypeEnum,
+  ResponseActiveInstallmentDto,
+  ResponseInstallmentItemDto,
+  ResponseInstallmentPaymentDetailDto,
+  ResponseOneTimePaymentDetailsDto,
+  ResponseTaxDetailItemizedDto,
+} from '../dtos/response.tax.dto'
+import { generateItemizedTaxDetail } from './helpers/tax.helper'
+
+dayjs.extend(utc)
+dayjs.extend(timezone)
+
+type ReplaceQrCodeWithGeneratorDto<T extends object> = {
+  [K in keyof T]: K extends 'qrCode' ? QrCodeGeneratorDto : T[K]
+}
+
+const bratislavaTimeZone = 'Europe/Bratislava'
+
+const stateHolidays: Dayjs[] = [
+  // 2025
+  dayjs.tz('2025-01-01', bratislavaTimeZone),
+  dayjs.tz('2025-01-06', bratislavaTimeZone),
+  dayjs.tz('2025-04-18', bratislavaTimeZone),
+  dayjs.tz('2025-04-21', bratislavaTimeZone),
+  dayjs.tz('2025-05-01', bratislavaTimeZone),
+  dayjs.tz('2025-05-08', bratislavaTimeZone),
+  dayjs.tz('2025-07-05', bratislavaTimeZone),
+  dayjs.tz('2025-08-29', bratislavaTimeZone),
+  dayjs.tz('2025-09-01', bratislavaTimeZone),
```

Comment:
this is tricky. While it is true, that 2025-09-01 (and following years too) are holidays, they are not bank holidays. Since we only use this array to determine a working day, then this should not be there, as it, in fact, is a working day despite it being a holiday.